### ISSUE TITLE
Support Github actions

### ISSUE THREAD

**Issue by ipa-mdl** [ros-industrial/industrial_ci#343]
I have already access to [Github actions](https://github.com/features/actions) as part of the beta program.
Starting today public repositories are now supported as well (just for owners in beta program).
It should not be too hard to come up with a first prototype.

**Comment by ipa-mdl** [comment]
I managed to configure a workflow with the help of our [Docker images](https://hub.docker.com/r/rosindustrial/ci/tags):

```
workflow "Industrial CI" {
  on = "push"
  resolves = ["melodic"]
}

action "melodic" {
  uses = "docker://rosindustrial/ci:melodic-bionic"
  args = ["-c", "git clone --depth 1 https://github.com/ros-industrial/industrial_ci.git /ici && /ici/ci.sh IN_DOCKER=1 TARGET_REPO_NAME=${GITHUB_REPOSITORY##*/}"]
  runs = "/bin/bash"
}
```

To test a different distro, just change the image name in `uses`.
We could simplify this by bundling a little helper script with our images.

**Comment by ipa-mdl** [comment]
> Any idea?

These images are outdated (https://github.com/ros-industrial/docker/pull/28)
Please try with `ros:kinetic-ros-core` (or alike) and the master branch of `industrial_ci`.
The latter is needed, because it works with the regular ROS images.

**Comment by gavanderhoorn** [comment]
@ipa-mdl: have there been any changes wrt Actions support?

Would you already be able to provide some best-practices?

**Comment by ipa-mdl** [comment]
@gavanderhoorn:

> have there been any changes wrt Actions support?

The version of @agutenkunst should basically work with the master branch and official ROS images.
I am unsure how we could support iteven further.
When I started to investigate, GitHub did not offer any option apart from bundling a Docker image. However, the system seems to have evolved since then.

> Would you already be able to provide some best-practices?

I did not use it for any practical purpose yet.

**Comment by ipa-mdl** [comment]
> However, the system seems to have evolved since then.

I played around a little bit.

```YAML
name: CI

on: [push]

jobs:
  industrial_ci:
    strategy:
      matrix:
        env:
          - ROS_DISTRO=melodic
          - ROS_DISTRO=melodic ROS_REPO=main
    name: ${{matrix.env}} 
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v1
    - run: git clone --quiet --depth=1 https://github.com/ros-industrial/industrial_ci.git .industrial_ci -b master
    - run: .industrial_ci/ci.sh ${{matrix.env}} TARGET_REPO_NAME="${GITHUB_REPOSITORY##*/}" TARGET_REPO_PATH="${GITHUB_WORKSPACE}"
```
works fairly well: https://github.com/ros-industrial/ros_canopen/commit/1ca03e19fd63a9d3ecbac6f4e74ef00d02355fb4/checks

We could simplify this by creating a specialized `github.sh` script that might even setup the folding.

**Comment by ipa-mdl** [comment]
closed by #437
